compiler/rustc_trait_selection/src/traits/error_reporting/on_unimplemented.rs

```diff
@@ -182,7 +182,9 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             // which we might want to be some sort of hygienic.
             //
             // Currently I'm leaving it for what I need for `try`.
-            if self.tcx.trait_of_item(item) == Some(trait_ref.def_id) {
+            if !self.tcx.is_impl_trait_in_trait(item)
+                && self.tcx.trait_of_item(item) == Some(trait_ref.def_id)
+            {
                 let method = self.tcx.item_name(item);
                 flags.push((sym::from_method, None));
```

Comment:
I'm pretty certain that nothing uses `from_method`. Can you verify this, and then just delete this whole if statement, and all references to `sym::from_method` from within the compiler?